vespa/deployment.py

```diff
@@ -550,127 +646,185 @@ def deploy(
+        :param submit_options: Optional dictionary of submit options, containing one or more of the following keys:
+            - "repository"
+            - "branch"
+            - "commit"
+            - "description"
+            - "authorEmail"
+            - "sourceUrl"
+            These will show up in the Vespa Cloud Console.
+
         """
         logging.warning(
-            "This feature is experimental and may fail in unexpected ways. Expect better support in future releases."
+            "Deploying to production is in beta and may fail in unexpected ways. Expect better support in future releases."
         )
-
-        if not disk_folder:
-            disk_folder = os.path.join(os.getcwd(), self.application)
-        self.application_package.to_files(disk_folder)
-
-        if self.application_package.deployment_config is None:
-            raise ValueError("'Prod deployment requires a deployment_config.")
-
-        self._start_prod_deployment(disk_folder)
-
+        if disk_folder is None:
+            if self.disk_folder is None:
+                disk_folder = os.path.join(os.getcwd(), self.application)
+            else:
+                disk_folder = self.disk_folder
+        if self.application_package is not None:
+            if self.application_package.deployment_config is None:
+                raise ValueError("Prod deployment requires a deployment_config.")
+            self.application_package.to_files(disk_folder)
+
+        self.build_no = self._start_prod_deployment(disk_folder, submit_options)
+        deployable_build_no = self._get_last_deployable(self.build_no)
+        if deployable_build_no != self.build_no:
+            print(
+                f"Returning last deployable build number: {deployable_build_no}",
+                file=self.output,
+            )
+            self.build_no = deployable_build_no
         deploy_url = "https://console.vespa-cloud.com/tenant/{}/application/{}/prod/deployment".format(
             self.tenant, self.application
         )
         print(f"Follow deployment at: {deploy_url}", file=self.output)
-
-        print("Waiting for monitoring...", file=self.output)
-        run_id = -1
-        updated = False
-        for _ in range(5):
-            new_run_id = self._get_latest_run_id(instance)  # This may or may not be
-            if new_run_id > run_id:
-                run_id = new_run_id
-                updated = True
-                break
-            sleep(4)
-        if not updated:
-            print(
-                "Not certain if this is the latest run id. Please monitor at: "
-                + deploy_url,
-                file=self.output,
+        return self.build_no
+
+    def _get_last_deployable(self, build_no: int) -> int:
+        # This is due to optimization that some builds will not be deployable (e.g if no diff from previous build)
+        # May take a few seconds for the build to show up in the deployment list
```

Comment:
Did you actually observe that? I suppose you _could_ see a delay, but I would think that was ultra-rare.